### PR TITLE
BG Queue Announcer FIX

### PR DESCRIPTION
**Pull request by wizzymore**
##### CHANGES PROPOSED:

-  Show to the player or to the world ( config based ) the queue status for the current bg

###### ISSUES ADDRESSED:

Closes 

##### TESTS PERFORMED:

- [x] Windows
- [x] Linux

##### HOW TO TEST THE CHANGES:

Go into worldserver.conf set `Battleground.QueueAnnouncer.Enable` to 1 and `Battleground.QueueAnnouncer.PlayerOnly ` first time to 0 and then to 1.
To test the changes queue to a bg ( not random ) and spam the button to join the queue.
Try it again in 30 seconds or spam it for 30+ seconds ( so it will write again and you can see that the spam protection works )
```
#
#    Battleground.QueueAnnouncer.Enable
#        Description: Announce battleground queue status to chat.
#        Default:     0 - (Disabled)
#                     1 - (Enabled)

Battleground.QueueAnnouncer.Enable = 1

#
#    Battleground.QueueAnnouncer.PlayerOnly
#        Description: Battleground queue announcement type.
#        Default:     0 - (System message, Anyone can see it)
#                     1 - (Private, Only queued players can see it)

Battleground.QueueAnnouncer.PlayerOnly = 0
```

##### KNOWN ISSUES AND TODO LIST:

- [x] It can be spammed at the moment

##### Target branch(es):

Master

### CODE REVIEW

**Comment by Viste** [comment]
why you split 3 other types of queue to one?

**Comment by Viste** [comment]
why you just remove arena queue announce?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 1]
```diff
@@ -194,21 +204,33 @@ GroupQueueInfo* BattlegroundQueue::AddGroup(Player* leader, Group* grp, PvPDiffi
                 if (!(*itr)->IsInvitedToBGInstanceGUID)
                     qHorde += (*itr)->Players.size();
 
-            // show queue status to player when joining queue
-            if (ginfo->BgTypeId == BATTLEGROUND_RB)
-                ChatHandler(leader->GetSession()).PSendSysMessage(LANG_BG_QUEUE_ANNOUNCE_80_RANDOM, bgName, q_min_level, q_max_level, qAlliance, qHorde);
-            else if (bgt->isArena())
```
Comment (Viste) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: why ?

Comment (wizzymore) [reply src/server/game/Battlegrounds/BattlegroundQueue.cpp]: I will fix it

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 2]
```diff
@@ -204,21 +204,21 @@ GroupQueueInfo* BattlegroundQueue::AddGroup(Player * leader, Group * grp, PvPDif
...
                 ChatHandler(leader->GetSession()).PSendSysMessage(LANG_BG_QUEUE_ANNOUNCE_SELF, bgName, q_min_level, q_max_level,
                     qAlliance, (MinPlayers > qAlliance) ? MinPlayers - qAlliance : (uint32)0, qHorde, (MinPlayers > qHorde) ? MinPlayers - qHorde : (uint32)0);
             }
-            // System message
-            else
+            // Show queue status to server (when joining battleground queue)
+            else if (!bgt->isArena())
             {
-                if (BGSpamProtection[leader->GetGUID()].last_queue == 0 && !bgt->isArena()) {
+                if (BGSpamProtection[leader->GetGUID()].last_queue == 0) {
```
Comment (Viste) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: code style! all { and } in other of code lines!

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 3]
```diff
@@ -204,21 +204,21 @@ GroupQueueInfo* BattlegroundQueue::AddGroup(Player * leader, Group * grp, PvPDif
...
                     BGSpamProtection[leader->GetGUID()].last_queue = sWorld->GetGameTime();
                     sWorld->SendWorldText(LANG_BG_QUEUE_ANNOUNCE_WORLD, bgName, q_min_level, q_max_level,
                         qAlliance + qHorde, MaxPlayers);
                 }
-                else if (sWorld->GetGameTime() - BGSpamProtection[leader->GetGUID()].last_queue >= 30 && !bgt->isArena()) {
+                else if (sWorld->GetGameTime() - BGSpamProtection[leader->GetGUID()].last_queue >= 30) {
                     BGSpamProtection[leader->GetGUID()].last_queue = 0;
```
Comment (Viste) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp]: same here

**Comment by comix1988** [comment]
I tested this fix on ubuntu and it works.
Now I can only spam twice in world every 30 seconds.
https://youtu.be/h6xkkvhAUgA

good job @wizzymore

**Comment by milestorme** [comment]
Are we merging this yet ? 7 days waiting to be merged

**Comment by XdevTLKWoW** [comment]
Can confirm it.Works excellent !

**Pull request merged**